components/issue-proposals.tsx

```diff
@@ -46,11 +48,21 @@ export default function IssueProposals({ numberProposals, issueId, amount }) {
             merge_id: i,
           })
           .then((values: Proposal) => {
-            values.isDisputed = BeproService.network.IsMergeDisputed({
-              issueId: issueId,
-              mergeId: i,
-            });
-            arrayProposals.push(values);
+            BeproService.network
```

Comment:
you don't need to enter in a "callback hell", you can make these return their value and follow on the first then scope like so:

```ts
for (let i = 0; i < numProposals; i++) {
  const merge = await BeproService.network.getMergeById({issue_id, merge_id});
  const isMergeDisputed = await BeproService.network.IsMergeDisputed({issue_id, merge_id});
  const mergeProposal = await GithubMicroservice.getMergeProposalIssue({issue_id, merge_id});
  
  merge.isDisputed = isDisputed;
  merge.pullRequestId = mergeProposal.pullRequestId;
  
  arrayProposals.push(merge)
}
```